### PR TITLE
Added some fog and 1 tutorial map

### PR DESCRIPTION
**Pull request by Bluntsord**
Added some fog with a fog shader for the main screen. Can be used for other maps.
Also made the first tutorial map to examplify stuff

### CODE REVIEW

**Comment by bharathcs** [comment]
What the. Ups la tc

File: game/src/environment/LevelMap/Tutorial.tscn [hunk 1]
```diff
@@ -1,7 +1,8 @@
-[gd_scene load_steps=5 format=2]
+[gd_scene load_steps=6 format=2]
 
 [ext_resource path="res://src/environment/LevelMap/core/LevelMap.tscn" type="PackedScene" id=1]
 [ext_resource path="res://src/environment/traps/Trap.tscn" type="PackedScene" id=2]
+[ext_resource path="res://src/ui/Menu/Fog.tscn" type="PackedScene" id=3]
```
Comment (bharathcs) [inline game/src/environment/LevelMap/Tutorial.tscn]: Do u wanna move Fog.tscn somewhere else or duplicate and put a copy in env/map/Fog.tscn? I think duplicating the file in this case q okay, makes sense too in case we have custom options for map.

File: game/src/environment/towers/core/Tower.tscn [hunk 2]
```diff
@@ -140,28 +137,18 @@ __meta__ = {
 [node name="CollisionShape2D" type="CollisionShape2D" parent="."]
 shape = SubResource( 1 )
 
-[node name="Range" type="Area2D" parent="."]
```
Comment (bharathcs) [inline game/src/environment/towers/core/Tower.tscn]: Why the changes here? 143-149

File: game/src/environment/towers/core/Tower.tscn [hunk 3]
```diff
@@ -8,105 +8,102 @@
 [sub_resource type="RectangleShape2D" id=1]
 extents = Vector2( 64, 64 )
 
-[sub_resource type="CircleShape2D" id=2]
```
Comment (bharathcs) [inline game/src/environment/towers/core/Tower.tscn]: Is this delete intended?

**Review by bharathcs — COMMENTED** [review]
I can't play the game due to edits you made in Tower.tscn 

I suspect you kept godot open while you pulled instead of quitting to project list, causing some mix up. Anyway, if the edits you made were not intended: hit this: https://github.com/bharathcs/quantum-defence/pull/62

The fog looks great btw, tested it after resetting tower and its looking good

File: game/src/environment/towers/core/Tower.tscn (hunk 2 above)
Comment (bharathcs) [reply game/src/environment/towers/core/Tower.tscn]: see https://github.com/bharathcs/quantum-defence/pull/61#pullrequestreview-699530283 and https://github.com/bharathcs/quantum-defence/pull/62

File: game/src/environment/towers/core/Tower.tscn (hunk 3 above)
Comment (bharathcs) [reply game/src/environment/towers/core/Tower.tscn]: see https://github.com/bharathcs/quantum-defence/pull/61#pullrequestreview-699530283 and https://github.com/bharathcs/quantum-defence/pull/62

File: game/src/environment/LevelMap/Tutorial.tscn (hunk 1 above)
Comment (Bluntsord) [reply game/src/environment/LevelMap/Tutorial.tscn]: yeah we can make dup it since each fog can be diff color and diff transparency

**Comment by bharathcs** [comment]
Looks good to me! I have extended ur branch here: If you are fine with the patch ( diff here: https://github.com/bharathcs/quantum-defence/pull/65/commits/751a6b7bb2694427b47ac0ed847560c64afb2c30 ) you can merge https://github.com/bharathcs/quantum-defence/pull/65 to main (all of these commits + my patch goes in).

**Review by bharathcs — APPROVED** [review]
Approved, but pls include my patch from https://github.com/bharathcs/quantum-defence/pull/65

**Comment by bharathcs** [comment]
Looks good to me! I have extended ur branch here: If you are fine with the patch ( diff here: https://github.com/bharathcs/quantum-defence/pull/65/commits/751a6b7bb2694427b47ac0ed847560c64afb2c30 ) you can merge https://github.com/bharathcs/quantum-defence/pull/65 to main (all of these commits + my patch goes in).
